sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_EventProcessor.md

```diff
@@ -0,0 +1,230 @@
+        {
+            { OffsetMetadataKey, data.Offset.ToString(CultureInfo.InvariantCulture) },
+        };
+
+        using MemoryStream emptyStream = new MemoryStream(Array.Empty<byte>());
+        await StorageContainer.GetBlobClient(checkpointBlob).UploadAsync(emptyStream, metadata: checkpointMetadata, cancellationToken: cancellationToken).ConfigureAwait(false);
+    }        
+}
+```
+
+## Using the Custom Processor
+
+The `AzureBlobStorageEventProcessor<TPartiton>` above is abstract because we did not implement any of the `On*` methods. By separating the logic for state management and event processing, we can leverage `AzureBlobStorageEventProcessor<TPartiton>` to build many different processors, each with their own business logic, by extending `AzureBlobStorageEventProcessor<TPartiton>` and implementing `OnProcessingEventBatchAsync` and `OnProcessingErrorAsync`.
+
+Here's a simple processor that just counts the number of events received in a batch and then creates a checkpoint so they are not processed again:
+
+```C# Snippet:EventHubs_Sample08_CustomProcessor
+class CustomProcessor : AzureBlobStorageEventProcessor<EventProcessorPartition>
+{
+    public CustomProcessor(int eventBatchMaximumCount, string consumerGroup, string fullyQualifiedNamespace, string eventHubName, TokenCredential credential, BlobContainerClient storageContainer, EventProcessorOptions options = null) : base(eventBatchMaximumCount, consumerGroup, fullyQualifiedNamespace, eventHubName, credential, storageContainer, options)
+    {
+    }
+
+    protected async override Task OnProcessingEventBatchAsync(IEnumerable<EventData> events, EventProcessorPartition partition, CancellationToken cancellationToken)
+    {
+        Console.WriteLine($"Received batch of {events.Count()} events for partition {partition.PartitionId}");
+
+        if (events.Any())
+        {
+            await CheckpointAsync(partition, events.Last(), cancellationToken);
+        }
+    }
+
+    protected async override Task OnProcessingErrorAsync(Exception exception, EventProcessorPartition partition, string operationDescription, CancellationToken cancellationToken)
+    {
+        if (partition != null)
+        {
+            Console.Error.WriteLine($"Exception on partition {partition.PartitionId} while performing {operationDescription}: {exception.Message}");
+        }
+        else
+        {
+            Console.Error.WriteLine($"Exception while performing {operationDescription}: {exception.Message}");
+        }        
+    }    
+}
+```
+
+This should look familiar if you've used `EventProcessorClient` before, but there are a few small differences you need to be aware of:
+
+- The processor works in batches, so when you construct it you need to specify the maximum number of events you'll want to process at once, using the `eventBatchMaximumCount` parameter. The `IEnumerable<EventData>` that is passed `OnProcessingEventBatchAsync` to will never contain more items than this maximum count.
```

Comment:
But it may often contain less.   It is not unusual (though it is confusing) to see uneven sizes despite having a good number of events in the partition; this is due to how the AMQP library's prefetch cache works and is sensitive to network speed and other resource use.    That tends to confuse the heck out of people.